internal/dbtools/membership_enumeration.go

```diff
@@ -230,6 +230,82 @@ func GetAllGroupMemberships(ctx context.Context, db *sql.DB, shouldPopulateAllMo
 	return enumeratedMemberships, nil
 }
 
+// HierarchyWouldCreateCycle returns true if a given new parent->member relationship would create a cycle in the database
+func HierarchyWouldCreateCycle(ctx context.Context, db *sql.DB, parentGroupID, memberGroupID string) (bool, error) {
+	hierarchies := make(map[string][]string)
+
+	hierarchyRows, err := models.GroupHierarchies().All(ctx, db)
```

Comment:
I don't think so, this is just getting all of the hierarchy relations, not any associated groups or users. Even a large hierarchical setup would probably be... tens to low hundreds? I think it would add a fair bit of complication to try to form a better query for getting only the ones that could potentially be involved in this new hypothetical relation forming a cycle (especially since this is an infrequent activity).